grpc.go

```diff
@@ -1442,3 +1442,15 @@ func (a *agentGRPC) getAgentDetails(ctx context.Context) *pb.AgentDetails {
 
 	return &details
 }
+
+func (a *agentGRPC) UploadFile(ctx context.Context, req *pb.UploadFileRequest) (*gpb.Empty, error) {
+	if err := os.MkdirAll(filepath.Dir(req.Path), os.FileMode(req.DirMode)); err != nil {
+		return emptyResp, err
+	}
+
+	if err := ioutil.WriteFile(req.Path, []byte(req.Content), os.FileMode(req.FileMode)); err != nil {
```

Comment:
Having the content as an array of bytes would prevent from the cast here, and would be more appropriate.